src/mono/mono/mini/intrinsics.c

```diff
@@ -1007,6 +1007,54 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			EMIT_NEW_BIALU_IMM (cfg, ins, OP_COMPARE_IMM, -1, dreg, 0);
 			EMIT_NEW_UNALU (cfg, ins, OP_ICGT, dreg, -1);
 			ins->type = STACK_I4;
+			return ins;
+		} else if (!strcmp (cmethod->name, "CreateSpan") && fsig->param_count == 1) {
+			MonoGenericContext* ctx = mono_method_get_context (cmethod);
+			g_assert (ctx);
+			g_assert (ctx->method_inst);
+			g_assert (ctx->method_inst->type_argc == 1);
+			MonoType* arg_type = ctx->method_inst->type_argv [0];
+			MonoType* t = mini_get_underlying_type (arg_type);
+			g_assert (!MONO_TYPE_IS_REFERENCE (t) && t->type != MONO_TYPE_VALUETYPE);
+
+			// The following relies on the previous instruction being an OP_MOVE. Specifically. one that
+			// is emitted from CEE_TDTOKEN as the last EMIT_NEW_TEMPLOAD, which has its inst_p1 set.
+			// Bail out and use the non-intrinsic variant if this is not satisfied.
+			MonoClassField* field = (MonoClassField*) args [0]->inst_p1;
+			if (args [0]->opcode != OP_VMOVE || !field || !field->type)
+					return NULL;
+
+			int alignment = 0;
+			int element_size = mono_type_size (t, &alignment);
+
+		#if G_BYTE_ORDER == G_LITTLE_ENDIAN
```

Comment:
```suggestion
#if G_BYTE_ORDER == G_LITTLE_ENDIAN
```

Reply:
I have moved the preprocessor directives.